### PR TITLE
ci: add `_LIBCPP_REMOVE_TRANSITIVE_INCLUDES` to TSAN (libc++) job

### PR DESCRIPTION
**Pull request by fanquake**
Add `_LIBCPP_REMOVE_TRANSITIVE_INCLUDES` to one of the libc++ CI jobs, to catch missing includes, that are otherwise hidden by transitive includes inside libc++. A more appropriate place for this might be the tidy job, but that does not use libc++.

See https://libcxx.llvm.org/DesignDocs/HeaderRemovalPolicy.html for more information.

### CODE REVIEW

**Review by maflcko — APPROVED** [review]
ACK ea038bd228284eee2840f8f0ec2eb58fd0ad38d8

This is useful to catch build failures before they hit users that compile from source.

File: ci/test/00_setup_env_native_tsan.sh [hunk 1]
```diff
@@ -11,4 +11,4 @@ export CI_IMAGE_NAME_TAG="docker.io/ubuntu:24.04"
 export PACKAGES="clang-18 llvm-18 libclang-rt-18-dev libc++abi-18-dev libc++-18-dev python3-zmq"
 export DEP_OPTS="CC=clang-18 CXX='clang++-18 -stdlib=libc++'"
 export GOAL="install"
-export BITCOIN_CONFIG="--enable-zmq CPPFLAGS='-DARENA_DEBUG -DDEBUG_LOCKORDER -DDEBUG_LOCKCONTENTION' --with-sanitizers=thread"
+export BITCOIN_CONFIG="--enable-zmq CPPFLAGS='-DARENA_DEBUG -DDEBUG_LOCKORDER -DDEBUG_LOCKCONTENTION -D_LIBCPP_REMOVE_TRANSITIVE_INCLUDES' --with-sanitizers=thread"
```
[the remark this answers is not in the document]
Comment (fanquake) [reply ci/test/00_setup_env_native_tsan.sh:14]: Thanks, had a look at this, and ran into issues with at least leveldb, which we'd first have to fix in the subtree, i.e:
```bash
In file included from leveldb/util/comparator.cc:14:
./leveldb/util/no_destructor.h:40:17: error: 'aligned_storage<8, 8>' is deprecated [-Werror,-Wdeprecated-declarations]
   40 |   typename std::aligned_storage<sizeof(InstanceType),
      |                 ^
leveldb/util/comparator.cc:71:47: note: in instantiation of template class 'leveldb::NoDestructor<leveldb::(anonymous namespace)::BytewiseComparatorImpl>' requested here
   71 |   static NoDestructor<BytewiseComparatorImpl> singleton;
      |                                               ^
/usr/lib/llvm-18/bin/../include/c++/v1/__type_traits/aligned_storage.h:90:8: note: 'aligned_storage<8, 8>' has been explicitly marked deprecated here
   90 | struct _LIBCPP_DEPRECATED_IN_CXX23 _LIBCPP_TEMPLATE_VIS aligned_storage {
      |        ^
/usr/lib/llvm-18/bin/../include/c++/v1/__config:1013:41: note: expanded from macro '_LIBCPP_DEPRECATED_IN_CXX23'
 1013 | #    define _LIBCPP_DEPRECATED_IN_CXX23 _LIBCPP_DEPRECATED
      |                                         ^
/usr/lib/llvm-18/bin/../include/c++/v1/__config:974:49: note: expanded from macro '_LIBCPP_DEPRECATED'
  974 | #      define _LIBCPP_DEPRECATED __attribute__((__deprecated__))
      |                                                 ^
1 error generated.
```
So might followup with this.

File: src/init.h [hunk 2]
```diff
@@ -7,6 +7,7 @@
 #define BITCOIN_INIT_H
 
 #include <any>
+#include <atomic>
 #include <memory>
 #include <string>
```
[the remark this answers is not in the document]
Comment (fanquake) [reply src/init.h:12]: Thanks, added.

**Comment by hebasto** [comment]
Ported to the CMake-based build system in https://github.com/hebasto/bitcoin/pull/281.

**Pull request merged**